src/Compilers/CSharp/Portable/Binder/Binder_Initializers.cs

```diff
@@ -115,6 +115,17 @@ internal struct ProcessedFieldInitializers
                                 parentBinder = parentBinder.GetFieldInitializerBinder(fieldSymbol);
 
                                 BoundFieldEqualsValue boundInitializer = BindFieldInitializer(parentBinder, fieldSymbol, initializerNode, diagnostics);
+
+                                if (!boundInitializer.Value.HasAnyErrors)
+                                {
+                                    var field = boundInitializer.Field;
+                                    if (field.Type.IsRefLikeType)
```

Comment:
It wasn't clear to me whether we should check `field.Type.IsRefLikeType` or `field.ContainingType.IsRefLikeType` or both.

Reply:
As you mentioned, the safe-to-escape of a `static` field is global.

I've added an item to the ref fields test plan to consider `ref struct` field initializers when ref fields is supported.